### PR TITLE
Expose addl tower api behavior

### PR DESCRIPTION
**Pull request by syncrou**

### CODE REVIEW

**Comment by syncrou** [comment]
@bdunne - This pull request exposes the get methods for JobTemplate, AdHocCommands

File: lib/ansible_tower_client/common.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+module AnsibleTowerClient
+  module Common
```
Comment (bdunne) [inline lib/ansible_tower_client/common.rb]: What do you think of making this a superclass of the other three called `Collection`?

File: lib/ansible_tower_client/common.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        body = JSON.parse(Api.get(body["next"]).body)
+        results += body["results"]
+      end
+
+      results.collect { |raw| new(raw) }
+    end
+
+    private
+
+    def mapping
```
Comment (bdunne) [inline lib/ansible_tower_client/common.rb]: Should this maybe be a constant like `ENDPOINT = "hosts".freeze` on each class?

File: lib/ansible_tower_client/ad_hoc_command.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  class AdHocCommand
+    extend Common
+    attr_reader :id
+    attr_reader :name
+    attr_reader :raw_ad_hoc_common_body
+
+    def initialize(raw_ad_hoc_command)
+      @id   = raw_ad_hoc_command["id"]
+      @name = raw_ad_hoc_command["name"]
+      @raw_ad_hoc_command_body = raw_ad_hoc_command
```
Comment (bdunne) [inline lib/ansible_tower_client/ad_hoc_command.rb]: I'm starting to think this should be `@raw_body` or `@raw_result`

**Comment by bdunne** [comment]
Nice work!

File: lib/ansible_tower_client/common.rb (hunk 1 above)
Comment (syncrou) [inline lib/ansible_tower_client/common.rb]: Cool - I like that idea.

File: lib/ansible_tower_client/ad_hoc_command.rb (hunk 3 above)
Comment (syncrou) [inline lib/ansible_tower_client/ad_hoc_command.rb]: Yes - the specific naming was turning into a bit of a hassle.

**Comment by syncrou** [comment]
@miq-bot add_label wip

**Comment by syncrou** [comment]
@bdunne - Please review

File: lib/ansible_tower_client/collection_methods.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        body = JSON.parse(Api.get(body["next"]).body)
+        results += body["results"]
+      end
+
+      results.collect { |raw| new(raw) }
+    end
+
+    def find(id)
+      body = JSON.parse(Api.get("#{mapping}/#{id}/").body)
+      raise NoMethodError if body['id'].nil?
```
Comment (syncrou) [inline lib/ansible_tower_client/collection_methods.rb]: I realize this should probably be a custom or Faraday based exception - I have NoMethodError error there now mostly as a placeholder. Wouldn't mind thoughts on this.

File: lib/ansible_tower_client/collection_methods.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      new(body)
+    end
+
+    private
+
+    def mapping
```
Comment (bdunne) [inline lib/ansible_tower_client/collection_methods.rb]: Can this instead be a constant `ENDPOINT` or `RESOURCE` or something on the individual class?

**Pull request merged**